File: src/components/JSONHandler/src/ALRPCObjectsImpl/SoftButtonCapabilities.cpp
#include "../include/JSONHandler/ALRPCObjects/SoftButtonCapabilities.h"
#include "SoftButtonCapabilitiesMarshaller.h"

/*
  interface	Ford Sync RAPI
  version	1.2 / 2.0O
  date		2011-05-17 / 2012-11-02
  generated at	Thu Nov 29 14:49:08 2012
  source stamp	Thu Nov 29 06:50:10 2012
  author	robok0der
*/



using namespace NsAppLinkRPC;

SoftButtonCapabilities::SoftButtonCapabilities(const SoftButtonCapabilities& c)
{
  *this=c;
}


bool SoftButtonCapabilities::checkIntegrity(void)
{
  return SoftButtonCapabilitiesMarshaller::checkIntegrity(*this);
}


SoftButtonCapabilities::SoftButtonCapabilities(void)
{
}



bool SoftButtonCapabilities::set_imageSupported(bool imageSupported_)
{
  imageSupported=imageSupported_;
  return true;
}

bool SoftButtonCapabilities::set_longPressAvailable(bool longPressAvailable_)
{
  longPressAvailable=longPressAvailable_;
  return true;
}

bool SoftButtonCapabilities::set_shortPressAvailable(bool shortPressAvailable_)
{
  shortPressAvailable=shortPressAvailable_;
  return true;
}

bool SoftButtonCapabilities::set_upDownAvailable(bool upDownAvailable_)
{
  upDownAvailable=upDownAvailable_;
  return true;
}




bool SoftButtonCapabilities::get_imageSupported(void) const
{
  return imageSupported;
}


bool SoftButtonCapabilities::get_longPressAvailable(void) const
{
  return longPressAvailable;
}


bool SoftButtonCapabilities::get_shortPressAvailable(void) const
{
  return shortPressAvailable;
}


bool SoftButtonCapabilities::get_upDownAvailable(void) const
{
  return upDownAvailable;
}

File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/SoftButtonCapabilities.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCStruct;
import com.ford.syncV4.proxy.constants.Names;

public class SoftButtonCapabilities extends RPCStruct {

    public SoftButtonCapabilities() { }
    public SoftButtonCapabilities(Hashtable hash) {
        super(hash);
    }
    public void setShortPressAvailable(Boolean shortPressAvailable) {
        if (shortPressAvailable != null) {
            store.put(Names.shortPressAvailable, shortPressAvailable);
        } else {
        	store.remove(Names.shortPressAvailable);
        }
    }
    public Boolean getShortPressAvailable() {
        return (Boolean) store.get( Names.shortPressAvailable);
    }
    public void setLongPressAvailable(Boolean longPressAvailable) {
        if (longPressAvailable != null) {
            store.put(Names.longPressAvailable, longPressAvailable);
        } else {
        	store.remove(Names.longPressAvailable);
        }
    }
    public Boolean getLongPressAvailable() {
        return (Boolean) store.get( Names.longPressAvailable);
    }
    public void setUpDownAvailable(Boolean upDownAvailable) {
        if (upDownAvailable != null) {
            store.put(Names.upDownAvailable, upDownAvailable);
        } else {
        	store.remove(Names.upDownAvailable);
        }
    }
    public Boolean getUpDownAvailable() {
        return (Boolean) store.get( Names.upDownAvailable);
    }
    public void setImageSupported(Boolean imageSupported) {
        if (imageSupported != null) {
            store.put(Names.imageSupported, imageSupported);
        } else {
        	store.remove(Names.imageSupported);
        }
    }
    public Boolean getImageSupported() {
        return (Boolean) store.get( Names.imageSupported);
    }
}
